recipes/qt/6.x.x/conanfile.py

```diff
@@ -867,10 +860,13 @@ def package(self):
         for target in targets:
             exe_path = None
             for path_ in [f"bin/{target}{extension}",
-                          f"lib/{target}{extension}"]:
+                          f"lib/{target}{extension}",
+                          f"libexec/{target}{extension}"]:
                 if os.path.isfile(os.path.join(self.package_folder, path_)):
                     exe_path = path_
                     break
+            else:
+                assert False, f"Could not find executable {target}{extension} in {self.package_folder}"
```

Comment:
wanted to test on my side but Qt 6.7.1 build failed, will try with a different version later. Testing with v1 and
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=15
os=Macos
os_build=Macos
[options]
[build_requires]
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
```
(Xcode 15.4)

Reply:
Hi! If using Conan 2.4, you might want to use this fix https://github.com/conan-io/conan-center-index/pull/24236 for it to compile (Or wait until it gets merged in a few hours/tomorrow and update this PR with the changes)

Reply:
in my case build error was from Qt, and that PR is about Windows :)
```
/Applications/Xcode-15.4.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/__memory/unique_ptr.h:66:19: error: invalid application of 'sizeof' to an incomplete type 'QQmlJS::Dom::BindingValue'
...
/Users/kambala/.conan/data/qt/6.6.3/_/_/source/src/qtdeclarative/src/qmldom/qqmldomelements_p.h:513:7: note: forward declaration of 'QQmlJS::Dom::BindingValue'
class BindingValue;
```

just built with `-o 'qt/*:essential_modules=False'` and all went fine, didn't face that error. Could you post exact conan command you issued and the exact assert message (i.e. which executable wasn't found)?

Also pipeline was green...

Reply:
> Also pipeline was green...

At the moment this is unfortunately misleading, because the recipe is currently set to build on CI with different options than a user would get, see: https://github.com/conan-io/conan-center-index/pull/23911

This is wrong and needs to be fixed - however because of the large amount of versions we are maintaining, this would result in very long CI builds.

Will try to fix this locally, the expectation is that with `-o 'qt/*:essential_modules=True'` the changes in this PR (which are otherwise great) should not result in failures

Reply:
@RubenRBS still unclear how you hit that assert and what was the exact message

Reply:
I get this too, with a simple
```
conan create . --version=6.7.1
```

keeping in mind that this causes `essential_modules=True` (by default)

Reply:
Why is this being changed from a warning to an assert in this PR?
On master it doesn't find this tool either `qmltestrunner `, but it is a warning.

Reply:
`qmltestrunner` is not a documented qt app as far as I can find, and it seems like its only built when certain conditions are met (including, potentially, the building of tests) - so I've added a commit to remove it from the list of checked executables, until we can work out if this is needed and how to fix this

as a side note, and I appreciate the changes on this PR existed previously, it is very difficult to review pull requests when there are changes that have nothing to do with the PR description :(

Reply:
Tried building with v2 and faced that error as well.

Also input from Eric:
> it is meant to avoid regressions, because this PR changed the folder in which executables are packaged
If this assert is triggered for executable qmltestrunner ,then someone should go take a look in the package folder to find where is is stored, and then add the path in this list https://github.com/conan-io/conan-center-index/pull/24193/files#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R862
> qmltestrunner may only be built when `-o qt/*:qtshadertools=True`

Reply:
`qmltestrunner` is not packaged - either before or after the changes in this PR, because it is not being built.  

The problem exists outside of the changes in this PR and it unrelated  - so let's address the package structure first.